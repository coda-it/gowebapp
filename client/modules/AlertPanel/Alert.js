import React from 'react';
class Alert extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false,
            hide: false,
        };
    }
    componentDidMount() {
        this.showTimeout = setTimeout(() => {
            this.setState({ show: true });
        }, 2000);
        this.hideTimeout = setTimeout(() => {
            this.setState({ hide: true });
        }, 4000);
    }
    componentWillUnmount() {
        if (this.showTimeout) {
            clearTimeout(this.showTimeout);
        }
        if (this.hideTimeout) {
            clearTimeout(this.hideTimeout);
        }
    }
    render() {
        const { type, children } = this.props;
        const { show, hide } = this.state;
        const showClass = show ? 'show' : '';
        const hideClass = hide ? 'hide' : '';
        const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;
        return React.createElement("div", { className: classes }, children);
    }
}
export default Alert;
//# sourceMappingURL=Alert.js.map