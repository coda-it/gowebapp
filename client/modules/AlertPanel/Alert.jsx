// @flow
import React from 'react';

type Props = {
  type: string,
  children: React$Element<any>,
};

type State = {
  show: boolean,
  hide: boolean,
};

class Alert extends React.Component<Props, State> {
  constructor() {
    super();

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // $FlowFixMe - this component will be transformed into typescript soon
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // $FlowFixMe - this component will be transformed into typescript soon
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // $FlowFixMe - this component will be transformed into typescript soon
    if (this.showTimeout) {
      // $FlowFixMe - this component will be transformed into typescript soon
      clearTimeout(this.showTimeout);
    }
    // $FlowFixMe - this component will be transformed into typescript soon
    if (this.hideTimeout) {
      // $FlowFixMe - this component will be transformed into typescript soon
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
