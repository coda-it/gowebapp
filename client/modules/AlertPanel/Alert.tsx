import React from 'react';

type Props = {
  type: string;
  children: React.ReactNode;
};

type State = {
  show: boolean;
  hide: boolean;
};

class Alert extends React.Component<Props, State> {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // @ts-ignore
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // @ts-ignore
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // @ts-ignore
    if (this.showTimeout) {
      // @ts-ignore
      clearTimeout(this.showTimeout);
    }
    // @ts-ignore
    if (this.hideTimeout) {
      // @ts-ignore
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
