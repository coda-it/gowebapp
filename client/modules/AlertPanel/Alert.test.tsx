import React from 'react';
import { render } from '@testing-library/react';
import Alert from './Alert';

describe('AlertPanel/Alert', () => {
  it('should render correctly', () => {
    const { container } = render(<Alert type="type-class">Message</Alert>);

    // @ts-ignore
    expect(container.firstChild).toMatchSnapshot();
  });
});
