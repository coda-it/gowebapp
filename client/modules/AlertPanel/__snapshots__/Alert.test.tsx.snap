// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPanel/Alert should render correctly 1`] = `
<div
  class="alert-panel__alert alert-panel__alert--type-class  "
>
  Message
</div>
`;
